rust-runtime/aws-smithy-types/src/byte_stream/bytestream_util.rs

```diff
@@ -230,15 +233,18 @@ impl FsBuilder {
 enum State {
     Unloaded(PathBuf),
     Loading(Pin<Box<dyn Future<Output = io::Result<File>> + Send + Sync + 'static>>),
-    Loaded(ReaderStream<io::Take<File>>),
+    Loaded {
```

Comment:
**question**: I assume this is the case but this gets reset correctly on a retry right?